File: Source/CSharp/Expressive/Expressive/Expressions/ConstantValueExpression.cs
using System.Collections.Generic;

namespace Expressive.Expressions
{
    internal class ConstantValueExpression : IExpression
    {
        private readonly ConstantValueExpressionType _expressionType;
        private readonly object _value;

        internal ConstantValueExpression(ConstantValueExpressionType type, object value)
        {
            _expressionType = type;
            _value = value;
        }

        #region IExpression Members

        public object Evaluate(IDictionary<string, object> variables)
        {
            return _value;
        }

        #endregion
    }

    internal enum ConstantValueExpressionType
    {
        Unknown,
        Integer,
        String,
        DateTime,
        Float,
        Boolean,
        Null,
        Double,
        Decimal,
        Long,
    }
}

File: Source/Java/Expressive/src/main/java/com/bijington/expressive/expressions/ConstantValueExpression.java
package com.bijington.expressive.expressions;

import java.util.Map;

/**
 * Created by shaun on 05/07/2016.
 */
public class ConstantValueExpression implements IExpression {
    //private final ConstantValueExpressionType _expressionType;
    private final Object _value;

    public ConstantValueExpression(Object value) {
        //_expressionType = type;
        _value = value;
    }

    @Override
    public Object evaluate(Map<String, Object> variables) { return _value; }
}
